internal/handler/command.go

```diff
@@ -68,28 +68,31 @@ func CommandHandler(vars map[string]string, body string, method string) (*dsMode
 		}
 
 		if strings.ToLower(method) == common.GetCmdMethod {
-			return execReadDeviceResource(&d, &dr)
+			return execReadDeviceResource(&d, &dr, queryParams)
 		} else {
 			appErr := execWriteDeviceResource(&d, &dr, body)
 			return nil, appErr
 		}
 	}
 
 	if strings.ToLower(method) == common.GetCmdMethod {
-		return execReadCmd(&d, cmd)
+		return execReadCmd(&d, cmd, queryParams)
 	} else {
 		appErr := execWriteCmd(&d, cmd, body)
 		return nil, appErr
 	}
 }
 
-func execReadDeviceResource(device *contract.Device, dr *contract.DeviceResource) (*dsModels.Event, common.AppError) {
+func execReadDeviceResource(device *contract.Device, dr *contract.DeviceResource, queryParams string) (*dsModels.Event, common.AppError) {
 	var reqs []dsModels.CommandRequest
 	var req dsModels.CommandRequest
 	common.LoggingClient.Debug(fmt.Sprintf("Handler - execReadCmd: deviceResource: %s", dr.Name))
 
 	req.DeviceResourceName = dr.Name
 	req.Attributes = dr.Attributes
+	if queryParams != "" {
+		req.Attributes["query"] = queryParams
```

Comment:
"query" is a special attribute here.  To prevent the attribute name conflict in some protocol, I suggest to use the full name such as "urlRawQuery" here.
Also, please put this string into internal/common/consts.go